Assets/Scripts/ObjectPool/Demo/ReturnDemo.cs

```diff
@@ -0,0 +1,18 @@
+using Features.Managers;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ReturnDemo : MonoBehaviour, IPoolableObject
+{
+    public void ResetValues()
+    {
+        Debug.Log("Test");
+    }
+
+    void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.G))
```

Comment:
The object can return itself?

You don't want this behavior since objects shouldn't be able to manage their own lifetime.